node/src/projects.ts

```diff
@@ -30,19 +30,36 @@ export function list(): string[] {
  * Open a project
  *
  * @param path Path to the project folder
- * @param subscriber A subscriber function that will receive published
- *                   events for the project
+ * @param projectEventsSubscriber A subscriber function that will receive `ProjectEvent`s
+ * @param fileEventsSubscriber A subscriber function that will receive `FileEvent`s
  * @return A project
  */
 export function open(
   folder: string,
-  subscriber?: (topic: string, event: unknown) => unknown
+  projectEventsSubscriber?: Subscriber,
+  fileEventsSubscriber?: Subscriber
```

Comment:
What you think about making the `subscriber` argument an object, leaving room for future growth?

Reply:
Actually, IMO we should be consistent with `documents` and have separate `subscribe` and `unsubscribe` functions (we also provide these functions for `errors`). Apart from consistency it gives us the flexibility for future growth and perhaps most importantly will allow porting to a Websocket JSON-RPC approach (where we can't send callbacks but we can still subscribe to particular project events and get notifications).